src/test/feature/lib/hdfs_config.h

```diff
@@ -0,0 +1,172 @@
+     */
+    void getActiveDatanodes(std::vector<std::string> &activedatanodes,
+                            std::vector<int> &port);
+
+    /**
+     * whether HDFS is in safe mode
+     * @return true if HDFS is in safe node
+     */
+    bool isSafemode();
+
+    /**
+     * get parameter value in ./etc/hdfs-client.xml or ./etc/hadoop/hdfs-site.xml according to parameter name
+     * @param parameterName, used to get parameter value
+     * @param conftype, get parameter value, 'hdfs' or 'HDFS' from ./etc/hdfs-client.xml, others from ./etc/hadoop/hdfs-site.xml
+     * @return parameter value
+     */
+    std::string getParameterValue(const std::string &parameterName, const std::string &conftype);
+    
+    /**
+     * get parameter value in ./etc/hadoop/hdfs-site.xml according to parameter name
+     * @param parameterName, used to get parameter value
+     * @return parameter value
+     */
+    std::string getParameterValue(const std::string &parameterName);
+
+    /**
+     * set parameter value in ./etc/hdfs-client.xml or ./etc/hadoop/hdfs-site.xml according to parameter name
+     * @param parameterName, parameter name which used to set parameter value
+     * @param parameterValue, parameter value which to be set
+     * @param conftype, get parameter value, 'hdfs' or 'HDFS' from ./etc/hdfs-client.xml, others from ./etc/hadoop/hdfs-site.xml
+     * @return true if succeeded
+     */
+    bool setParameterValue(const std::string &parameterName, const std::string &parameterValue, const std::string &conftype);
+
+    /**
+     * set parameter value in ./etc/hadoop/hdfs-site.xml according to parameter name
+     * @param parameterName, parameter name which used to set parameter value
+     * @param parameterValue, parameter value which to be set
+     * @return true if succeeded
+     */
+    bool setParameterValue(const std::string &parameterName, const std::string &parameterValue);
+
+  private:
+    /**
+     * @return yarn user
+     */
+    std::string getHdfsUser();
+    /**
+     * load key-value parameters in ./etc/hdfs-client.xml
+     * @return true if succeeded
+     */
+    bool LoadFromHawqConfigFile();
+
+    /**
+     * load key-value parameters in ./etc/hadoop/hdfs-site.xml
+     * @return true if succeeded
+     */
+    bool LoadFromHdfsConfigFile();
+
+    /**
+     * get HDFS active or standby namenode information in HA mode according to the namenodetype
+     * @param namenodetype, used to specify active or standby namenode information
+     * @param namenode, namenode hostname reference which will be set
+     * @param port, namenode port reference which will be set
+     * @return true if getHANamenode succeeded
+     */
+    bool getHANamenode(const std::string &namenodetype,
+                       std::string &namenode,
+                                   int &port);
+
+  private:
+    std::unique_ptr<XmlConfig> hawqxmlconf;
+    std::unique_ptr<XmlConfig> hdfsxmlconf;
+    hawq::test::PSQL psql;
```

Comment:
You don't need psql object in this class. Remember modifying the construction too. Same problem also exists in `YarnConfig`.